### PR TITLE
Replace infinite scroll with pagination for actions page 3

### PR DESCRIPTION
**Pull request by Kinlaw**
- Replaces infinite scroll with pagination on actions page 3
- Adds really select all to allow playbook creation for all actions #203 
- Refactors actions service to request page-by-page data instead of querying for all results at once
- Changes system modal call to use inventory service so that all system modal calls are using the same function.

### CODE REVIEW

File: app/js/components/actions/actionsRule/actionsRule.jade [hunk 1]
```diff
@@ -24,103 +24,110 @@
...
+            thead
+              tr
+                th(style='width: 30px')
+                  input.action-checkbox(
+                    tooltip='Select All',
+                    type="checkbox",
+                    ng-indeterminate="checkboxes.indeterminate",
+                    ng-model='checkboxes.checked',
+                    ng-change='selectAll(); checkboxes.checkboxChecked(checkboxes.checked, ruleSystems)')
+                th.sortable(style='width: 27%;', ng-class="predicate | sortClass:'toString':reverse", ng-click="sort('toString')")
```
Comment (jharting) [inline app/js/components/actions/actionsRule/actionsRule.jade]: This should be `system_type_id` instead of `toString`

**Comment by jharting** [comment]
System selection seems to re-set when I switch to a different page. For example, I cannot select first system on page 1 and first system on page 2 and create a plan out of it. Is that intended? The tables in planner do not reset selection on page change.

**Comment by Kinlaw** [comment]
@jharting good catch on the system type sort. 

As of now it is intentional to not keep track of systems from previous pages unless the user selects all systems. I switched the actions page 3 over to loading only the systems of the current page unless the user selects all systems. In the case where the user selects all systems, I load all systems and we can pass all of those to the maintenance modal.

The way the api is currently written there's just not a good way to keep up with partial results. We could store the systems in a separate array each time we load a page. In that case we would have to traverse the array each time we loaded a page and verify that the systems aren't already in that array. That task would get progressively more intense as the array grows and eventually would outweigh the benefit of loading the results page-by-page. If we change the way pagination works on the backend to support an offset param and a number_of_results param then if a user jumps from page 1 to page 3, I can query for page 2 - 3 results and just gather a continuous list of systems which would avoid the need to traverse the array each time that I grab a page. I would know based on the size of the array if I had already loaded the current page of results into.

If we just keep track of the systems selected we would have to traverse the systems on the current page each time a checkbox is checked to see if each system is checked or unchecked. Then we would have to traverse the systemsSelected array to remove/add each of those systems. That could probably work. What do you think @jharting? Should I implement it that way before merging this in?

**Comment by jharting** [comment]
@Kinlaw I see. I was not aware it gets this complicated. I am going to merge this as is. Feel free to either implement the proposal on top of this or leave it and wait for feedback.

**Pull request merged**